spark/src/main/scala/io/delta/tables/execution/DeltaTableOperations.scala

```diff
@@ -112,6 +114,48 @@ trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>
     toDataset(sparkSession, restore)
   }
 
+  protected def executeClone(
+      table: DeltaTableV2,
+      target: String,
+      replace: Boolean,
+      properties: Map[String, String],
+      versionAsOf: Option[Long] = None,
+      timestampAsOf: Option[String] = None): DeltaTable = withActiveSession(sparkSession) {
+    val sourceIdentifier = table.getTableIdentifierIfExists.map(id =>
+      Identifier.of(id.database.toArray, id.table))
+    val sourceRelation = DataSourceV2Relation.create(table, None, sourceIdentifier)
+
+    val pathBased = new Path(target).isAbsolute()
+    val targetIdentifier = if (pathBased) s"delta.`$target`" else target
+    val targetRelation = UnresolvedRelation(
+      sparkSession.sessionState.sqlParser.parseTableIdentifier(targetIdentifier))
+
+    val maybeTimeTravelSource = if (timestampAsOf.isDefined || versionAsOf.isDefined) {
+      TimeTravel(
+        sourceRelation,
+        timestampAsOf.map(Literal(_)),
+        versionAsOf,
+        None
+      )
+    } else {
+      sourceRelation
+    }
+
+    val clone = CloneTableStatement(maybeTimeTravelSource, targetRelation, false, replace, true,
+      properties.toMap, None)
+
+    val qe = sparkSession.sessionState.executePlan(clone)
+
+    // call `QueryExecution.toRDD` to trigger the execution of commands.
+    SQLExecution.withNewExecutionId(qe, Some("clone delta table"))(qe.toRdd)
+
+    if (pathBased) {
+      DeltaTable.forPath(sparkSession, target)
```

Comment:
```suggestion
      io.delta.tables.DeltaTable.forPath(sparkSession, target)
```

Since I realized there is also https://github.com/delta-io/delta/blob/da5a5d29b6f00265439bc84020580080a2d4568f/spark/src/main/scala/org/apache/spark/sql/delta/DeltaTable.scala#L4

Reply:
I don't understand why you linked a line from the license here or what that has to do with anything. `DeltaTable` is already used directly in `trait DeltaTableOperations extends AnalysisHelper { self: DeltaTable =>`

Reply:
That is to clearly differentiate between `org.apache.spark.sql.delta.DeltaTable` and `io.delta.tables.DeltaTable`

Reply:
Had to change the existing usage too